scalafmt-sbt/src/main/scala/org/scalafmt/sbt/ScalafmtPlugin.scala

```diff
@@ -1,25 +1,175 @@
+
+  private def checkSources(
+      sources: Seq[File],
+      config: ScalafmtConfig,
+      log: Logger
+  ): Boolean = {
+    val res = withFormattedSources(sources, config)(
+      (file, e) => {
+        log.error(s"Error in ${file.toString}: $e")
+        false
+      },
+      (file, input, output) => {
+        val diff = input != output
+        if (diff) {
+          throw new MessageOnlyException(
+            s"${file.toString} isn't formatted properly!")
+        }
+        diff
+      }
+    ).flatten.forall(x => x)
+    PlatformTokenizerCache.megaCache.clear()
+    res
+  }
+
+  private lazy val sbtSources = thisProject.map(
+    proj => {
+      val rootSbt =
+        BuildPaths.configurationSources(proj.base).filterNot(_.isHidden)
+      val projectSbt =
+        (BuildPaths.projectStandard(proj.base) * GlobFilter("*.sbt")).get
+          .filterNot(_.isHidden)
+      rootSbt ++ projectSbt
+    }
+  )
+  private lazy val projectSources = thisProject.map(proj =>
+    (BuildPaths.projectStandard(proj.base) * GlobFilter("*.scala")).get)
+
+  lazy val scalafmtConfigSettings: Seq[Def.Setting[_]] = Seq(
+    scalafmt := formatSources(
+      (unmanagedSources in scalafmt).value,
+      scalaConfig.value,
+      streams.value.log),
+    scalafmtSbt := {
+      formatSources(sbtSources.value, sbtConfig.value, streams.value.log)
+      formatSources(projectSources.value, scalaConfig.value, streams.value.log)
+    },
+    scalafmtCheck :=
+      checkSources(
+        (unmanagedSources in scalafmt).value,
+        scalaConfig.value,
+        streams.value.log),
+    scalafmtSbtCheck := {
+      checkSources(sbtSources.value, sbtConfig.value, streams.value.log)
+      checkSources(projectSources.value, scalaConfig.value, streams.value.log)
+    },
+    compileInputs in compile := Def.taskDyn {
+      val task =
+        if (scalafmtOnCompile.value) scalafmt in resolvedScoped.value.scope
+        else Def.task(())
+      val previousInputs = (compileInputs in compile).value
+      task.map(_ => previousInputs)
+    }.value,
+    scalafmtOnly := {
+      val files = spaceDelimited("<files>").parsed
+      val absFiles = files.flatMap(fileS => {
+        Try { IO.resolve(baseDirectory.value, new File(fileS)) } match {
+          case Failure(e) =>
+            streams.value.log.error(s"Error with $fileS file: $e")
+            None
+          case Success(file) => Some(file)
+        }
+      })
+
+      val scalaFiles = absFiles.filter(_.toString.endsWith(".scala"))
+      formatSources(scalaFiles, scalaConfig.value, streams.value.log)
+      val sbtFiles = absFiles.filter(_.toString.endsWith(".sbt"))
+      formatSources(sbtFiles, sbtConfig.value, streams.value.log)
+    }
+  )
+
+  override def projectSettings: Seq[Def.Setting[_]] =
+    Seq(Compile, Test).flatMap(inConfig(_)(scalafmtConfigSettings))
+
+  override def buildSettings: Seq[Def.Setting[_]] = Seq(
+    scalafmtConfig := (baseDirectory in ThisBuild).value / ".scalafmt.conf"
```

Comment:
Will this fail if the file does not exist? I think sbt-scalafmt should work fine out of the box without any .scalafmt.conf, do we have a test for this?

Reply:
I can use `default` config when this file doesn't exist. I first used `Option[File]` with default `None`, but changed it thinking `.scalafmt` file **should** exists.

Reply:
I often use scalafmt on smaller personal projects with the default config without making a .scalafmt file.